### PR TITLE
Add ButtonStyle.premium

### PR DESCRIPTION
**Pull request by DA-344**
## Summary

DDevs Docs PR: https://github.com/discord/discord-api-docs/pull/6875

Removes ``InteractionResponse.require_premium`` in favor of ``ButtonStyle.premium`` like button components.

## Checklist

- [ ] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by AbstractUmbra** [comment]
This is *not* a breaking change, for what it's worth.

No contract has been broken since `.require_premium` never made it to a published release. I would argue that this checkbox could be removed from this PR.
Breaking changes do not happen between major version releases - if this is actually a breaking change, it will need to wait until 3.x!

File: discord/enums.py [hunk 1]
```diff
@@ -643,6 +644,7 @@ class ButtonStyle(Enum):
     green = 3
     red = 4
     url = 5
+    gradient = 6
```
Comment (Rapptz) [inline discord/enums.py:647]:
```suggestion
```

File: discord/ui/button.py [hunk 2]
```diff
@@ -77,6 +77,8 @@ class Button(Item[V]):
         like to control the relative positioning of the row then passing an index is advised.
         For example, row=1 will show up before row=2. Defaults to ``None``, which is automatic
         ordering. The row number must be between 0 and 4 (i.e. zero indexed).
+    sku_id: Optional[:class:`int`]
+        The SKU ID this button sends you to. Can't be combined with ``url``.
```
Comment (Rapptz) [inline discord/ui/button.py:81]: Missing versionadded.

File: docs/interactions/api.rst [hunk 3]
```diff
@@ -353,6 +356,9 @@ Enumerations
     .. attribute:: url
 
         An alias for :attr:`link`.
+    .. attribute:: gradient
+
+        An alias for :attr:`premium`
```
Comment (Rapptz) [inline docs/interactions/api.rst:361]:
```suggestion
```

File: docs/interactions/api.rst [hunk 4]
```diff
@@ -334,7 +334,10 @@ Enumerations
     .. attribute:: link
 
         Represents a link button.
+    .. attribute:: premium
 
+        Represents a gradient button denoting that buying a SKU is
+        required to perform this action.
```
Comment (Rapptz) [inline docs/interactions/api.rst:340]: Missing versionadded.

File: discord/ui/button.py [hunk 5]
```diff
@@ -202,6 +210,16 @@ def emoji(self, value: Optional[Union[str, Emoji, PartialEmoji]]) -> None:
         else:
             self._underlying.emoji = None
 
+    @property
+    def sku_id(self) -> Optional[int]:
+        """Optional[:class:`int`]: The SKU ID this button sends you to."""
```
Comment (Rapptz) [inline discord/ui/button.py:215]: Missing versionadded.

File: discord/components.py [hunk 6]
```diff
@@ -179,6 +179,7 @@ class Button(Component):
         'disabled',
         'label',
         'emoji',
+        'sku_id',
```
Comment (Rapptz) [inline discord/components.py:182]: Missing documentation + versionadded.

**Review by Rapptz — CHANGES_REQUESTED** [review]

**Review by Rapptz — APPROVED** [review]

**Pull request merged**